components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java

```diff
@@ -87,6 +87,11 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
             }
             RecoveryUtil.handleBadRequest(e.getMessage(), e.getErrorCode());
         } catch (IdentityRecoveryException e) {
+            if (e.getCause() != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
+                    e.getCause().getMessage())) {
+                LOG.error(e.getCause().getMessage(), e.getCause());
+                RecoveryUtil.handleBadRequest(e.getCause().getMessage(), Constants.ERROR_CODE_EMAIL_NOT_FOUND);
+            }
             RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, e.getErrorCode(), LOG, e);
         } catch (Throwable throwable) {
             if (throwable != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
```

Comment:
It is handled here right? How the exception related to email not found changed now as IdentityRecoveryException?

Reply:
Seems the type of the exception throwing has been changed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/151. Also while fixing check recover username flow as well.

Reply:
removed the email not found error checking from Throwable catch block.

In user name recovery flow also same issue is there. Added a fix to that as well.